File: backend/event/src/utils/logger.py
# src/utils/logger.py

"""
Logger Utility

Configures application-wide logging format and log levels.

Author: Vansh Patel (remotevansh@gmail.com)
Last Updated: February 19, 2025
"""

import logging

# Configure logging
logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")

# Create logger instance
logger = logging.getLogger("event-service")

File: backend/dbserver/src/utils/logger.ts
// src/utils/logger.ts
/**
    Logger Utility

    Configures and exports a Winston logger for structured logging.

    Features:
    - Logs messages in JSON format with timestamps.
    - Supports logging to both console and a file (`logs/app.log`).

    Dependencies:
    - winston (Logging library)

    Author: Vansh Patel (remotevansh@gmail.com)  
    Date: February 10, 2025  
 */
import winston from "winston";

const logFormat = winston.format.combine(
    winston.format.timestamp(),
    winston.format.json()
);

const logger = winston.createLogger({
    level: "info",
    format: logFormat,
    transports: [
        new winston.transports.Console(),
        new winston.transports.File({ filename: "logs/app.log" })
    ],
});

export { logger };
